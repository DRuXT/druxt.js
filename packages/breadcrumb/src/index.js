import { DruxtBreadcrumbModule } from './module'

/**
 * The Nuxt.js module function.
 *
 * Installs the module functionality in a Nuxt.js frontend.
 *
 * @type {Function}
 * @exports default
 * @name DruxtBreadcrumbModule
 * @see {@link ./module|DruxtBreadcrumbModule}
 */
export default DruxtBreadcrumbModule

/**
 * Vue.js Mixin.
 *
 * @type {object}
 * @exports DruxtBreadcrumbMixin
 * @see {@link ./mixins/breadcrumb|DruxtBreadcrumbMixin}
 */
 export { DruxtBreadcrumbMixin } from './mixins/breadcrumb'
