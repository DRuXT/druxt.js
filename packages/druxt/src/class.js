/**
 * Druxt utility class.
 *
 * @deprecated
 * @private
 */
class DruxtClass {
  /**
   * Class constructor.
   *
   * @example @lang js
   * import { DruxtClass } from 'druxt'
   * const druxt = new DruxtClass()
   */
  constructor() {}

  /**
   * Get component data from available options.
   *
   * @param {object} vm - The DruxtJS module Vue.js component.
   * @param {string[]} options - The component naming options.
   * @param {boolean} [all=false] - Returns all options if true, else only globally registered options.
   * @param {string} [prefix] - A string to prefix all components.
   *
   * @returns {WrapperComponents}
   */
  getComponents(vm, options, all = false, prefix) {
    const results = []
    const unique = {}

    options
      // Filter out incorrectly typed items.
      .filter(item => Array.isArray(item))

      // Process each item.
      .map(item => {
        const variants = []

        item.map(string => {
          const parts = variants.length ? [...variants[0].parts] : []
          parts.push(string)
          const clone = [...parts]

          // Attach prefix as required.
          if (typeof prefix !== 'string' && (prefix !== false || typeof prefix === 'undefined') && ((vm || {}).$options || {}).name) {
            prefix = vm.$options.name.match(/[A-Z][a-z]+/g).map(word => word.toLowerCase()).join('-')
          }

          if (prefix) {
            clone.unshift(prefix)
          }

          // Generate component name values.
          const kebab = clone.map(string => string.toLowerCase().replace(/--|_/g, '-')).join('-')
          const pascal = kebab.replace(/((\b|[^a-zA-Z0-9]+)[a-zA-Z0-9])/gi, (match, p1, p2) => match.toUpperCase().replace(p2, ''))

          // Check if component is globally registered.
          let global = false
          for (const name of [kebab, pascal]) {
            if (typeof (((vm || {}).$options || {}).components || {})[name] !== 'undefined') {
              global = true
              break
            }
          }

          variants.unshift({ global, kebab, parts, pascal, prefix })
        })

        // Add variants to results.
        variants.map(variant => {
          // Ensure unique results.
          if (unique[variant.pascal]) {
            return
          }
          unique[variant.pascal] = true

          results.push(variant)
        })
      })

    // Return globally registered components or all.
    return results
      .filter(option => option.global || !!all)
      .sort((a, b) => b.parts.length - a.parts.length)
  }

  /**
   * Get the Druxt module data from the referenced component.
   *
   * @returns {ModuleData}
   */
  async getModuleData(vm) {
    if (typeof ((vm || {}).$options || {}).druxt !== 'function') {
      return false
    }

    const moduleData = await vm.$options.druxt({ vm })

    if ((vm.$options || {}).name) {
      moduleData.name = vm.$options.name.match(/[A-Z][a-z]+/g).map(word => word.toLowerCase()).join('-')
    }

    return moduleData
  }
}

export { DruxtClass }
