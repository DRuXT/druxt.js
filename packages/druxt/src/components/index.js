/**
 * The Vue.js Druxt component.
 *
 * @type {object}
 * @exports Druxt
 * @name Druxt
 * @see {@link ./Druxt|API documentation}
 * @see {@link /guide/#the-druxt-component|Guide}
 *
 * @example @lang js <caption>Register globally</caption>
 * import Vue from 'vue'
 * import { Druxt } from 'druxt'
 *
 * Vue.component(Druxt)
 *
 * @example @lang vue <caption>Register locally</caption>
 * <template>
 *   <Druxt :module="module" />
 * </template>
 *
 * <script>
 * import { Druxt } from 'druxt'
 * export default {
 *   components: { Druxt }
 * }
 * </script>
 */
 export { default as Druxt } from './Druxt.vue'

 /**
 * The DruxtModule base Vue.js component.
 *
 * @type {object}
 * @exports DruxtModule
 * @name DruxtModule
 * @see {@link ./DruxtModule|API documentation}
 *
 * @example @lang vue
 * <script>
 * import DruxtModule from 'druxt/dist/components/DruxtModule.vue'
 * export default {
 *   name: 'MyDruxtModule',
 *   extends: DruxtModule,
 *   druxt: {
 *     componentOptions: () => ([['wrapper']]),
 *     propsData: (ctx) => ({ prop: ctx.prop }),
 *   }
 * }
 * </script>
 */
export { default as DruxtModule } from './DruxtModule.vue'

/**
 * The default Druxt module wrapper Vue.js component.
 *
 * @type {object}
 * @exports DruxtWrapper
 * @name DruxtWrapper
 * @see {@link ./DruxtWrapper|API documentation}
 * @see {@link /guide/#the-druxt-component|Guide}
 * @private
 */
 export { default as DruxtWrapper } from './DruxtWrapper.vue'
