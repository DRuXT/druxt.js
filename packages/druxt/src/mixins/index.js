/**
 * Vue.js Mixin to add support for the Wrapper system to a Druxt module.
 *
 * @type {object}
 * @exports DruxtComponentMixin
 *
 * @deprecated
 * @private
 */
 export { DruxtComponentMixin } from './component'
