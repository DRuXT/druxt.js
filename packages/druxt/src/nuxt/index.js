import chalk from 'chalk'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { join, normalize, resolve } from 'path'
import { DruxtClient } from '../client'
import meta from '../../package.json'

/**
 * Nuxt module function to install Druxt.
 *
 * @param {ModuleOptions} moduleOptions - DruxtJS module options.
 *
 * @example <caption>Nuxt configuration with module options</caption> @lang js
 * module.exports = {
 *   modules: [
 *     ['druxt', { baseUrl: 'https://demo-api.druxtjs.org' }]
 *   ]
 * }
 *
 * @example <caption>Nuxt configuration with root level options</caption> @lang js
 * module.exports = {
 *   modules: [
 *     'druxt'
 *   ],
 *   druxt: {
 *     baseUrl: 'https://demo-api.druxtjs.org'
 *   }
 * }
 */
const DruxtNuxtModule = async function (moduleOptions = {}) {
  const options = {
    endpoint: '/jsonapi',
    ...moduleOptions,
    ...(this.options || {}).druxt,
  }

  // Normalize slashes.
  this.options.baseUrl = options.baseUrl = options.baseUrl.endsWith('/') ? options.baseUrl.slice(0, -1) : options.baseUrl
  this.options.endpoint = options.endpoint = options.endpoint.startsWith('/') ? options.endpoint : `/${options.endpoint}`

  const druxt = new DruxtClient(options.baseUrl, {
    ...options,
    // Disable API Proxy, as Proxies aren't available at build.
    proxy: { ...options.proxy || {}, api: false },
  })

  // Nuxt proxy integration.
  if (options.proxy) {
    const proxies = {}

    // Enable proxying of the API endpoint.
    // This is primarily used to avoid CORS errors.
    if ((options.proxy || {}).api) {
      // Main API Endpoint.
      proxies[options.endpoint] = options.baseUrl

      // Langcode prefixed API endpoints.
      const languageResourceType = 'configurable_language--configurable_language'
      if (((await druxt.getIndex(languageResourceType)) || {}).href) {
        const query = new DrupalJsonApiParams().addFields(languageResourceType, ['drupal_internal__id'])
        const languages = (await druxt.getCollectionAll(languageResourceType, query) || [])
          .map((o) => o.data)
          .flat()
          .filter((o) => !['und', 'zxx'].includes(((o || {}).attributes || {}).drupal_internal__id))
          .map((o) => o.attributes.drupal_internal__id)
        for (const langcode of languages) {
          proxies[`/${langcode}${options.endpoint}`] = options.baseUrl
        }
      }

      // Decoupled Router Endpoint.
      proxies['/router/translate-path'] = options.baseUrl
    }

    // Enable proxying of the Drupal site files.
    if ((options.proxy || {}).files) {
      const filesPath = typeof options.proxy.files === 'string' ? options.proxy.files : 'default'
      proxies[`/sites/${filesPath}/files`] = options.baseUrl
    }

    // If there are existing proxy settings, merge in the appropriate format.
    if (this.options.proxy) {
      if (Array.isArray(this.options.proxy)) {
        this.options.proxy = [
          ...this.options.proxy,
          ...Object.keys(proxies).map((path) => `${options.baseUrl}${path}`)
        ]
      }
      else {
        this.options.proxy = {
          ...this.options.proxy,
          ...proxies
        }
      }
    }
    // Otherwise just set the the required proxies.
    else {
      this.options.proxy = proxies
    }

    // Enable the Proxy module.
    this.addModule('@nuxtjs/proxy')
  }

  // Install the @nuxtjs/axios module.
  if (!options.axios) {
    this.options.axios = {
      baseURL: options.baseUrl,
      proxy: !!(options.proxy || {}).api,
      ...this.options.axios
    }
    this.addModule('@nuxtjs/axios')
  }

  // Register components directories.
  this.nuxt.hook('components:dirs', dirs => {
    dirs.push({ path: join(__dirname, 'components') })
  })

  // Add plugin.
  this.addPlugin({
    src: resolve(__dirname, '../templates/plugin.js'),
    fileName: 'druxt.js',
    options
  })

  // Make the $axios and $druxt plugins the first to load.
  const extendPlugins = this.options.extendPlugins
  this.options.extendPlugins = (plugins) => {
    // Run the user defined extendPlugins function if defined.
    plugins = typeof extendPlugins === 'function' ? extendPlugins(plugins) : plugins

    // Extract the $axios plugin.
    const axiosIndex = plugins.findIndex(({ src }) => src === normalize(`${this.options.buildDir}/axios.js`))
    const axiosPlugin = plugins[axiosIndex]
    plugins.splice(axiosIndex, 1)

    // Extract the $druxt plugin.
    const druxtIndex = plugins.findIndex(({ src }) => src === normalize(`${this.options.buildDir}/druxt.js`))
    const druxtPlugin = plugins[druxtIndex]
    plugins.splice(druxtIndex, 1)

    // Re-add the plugins.
    plugins = [axiosPlugin, druxtPlugin, ...plugins]

    return plugins
  }

  // Add Vuex plugin.
  this.addPlugin({
    src: resolve(__dirname, '../templates/store.js'),
    fileName: 'store/druxt.js',
    options
  })

  // Enable Vuex Store.
  this.options.store = true

  // Enable components auto-discovery by default.
  this.options.components = this.options.components ?? true

  // Add CLI badge.
  this.options.cli.badgeMessages.push(`${chalk.blue.bold('Druxt')} @ v${meta.version}`)
  this.options.cli.badgeMessages.push(`${chalk.bold('API:')} ${chalk.blue.underline(options.baseUrl + options.endpoint)}`)

  // Development mode features.
  if (this.options.dev) {
    // Add the template stubber server middleware.
    this.addServerMiddleware({
      path: '/_druxt/template',
      handler: 'druxt/dist/server-middleware/template.mjs'
    })

    // Add the Vue devtools plugin.
    this.addPlugin({
      src: resolve(__dirname, '../dist/plugins/devtools.mjs'),
      fileName: 'druxt-devtools.js'
    })
  }

  // Nuxt Storybook.
  const self = this
  this.nuxt.hook('storybook:config', async ({ stories }) => {
    // Druxt README story.
    self.addTemplate({
      src: resolve(__dirname, '../templates/stories/README.stories.mdx'),
      fileName: 'stories/druxt-README.stories.mdx',
    })
    stories.push(resolve(self.options.buildDir, './stories/druxt-README.stories.mdx'))

    // Druxt custom module story.
    self.addTemplate({
      src: resolve(__dirname, '../templates/stories/druxt-module.stories.mdx'),
      fileName: 'stories/druxt-module.stories.mdx',
    })
    stories.push(resolve(self.options.buildDir, './stories/druxt-module.stories.mdx'))

    // DruxtDebug component story.
    self.addTemplate({
      src: resolve(__dirname, '../templates/stories/druxt-debug.stories.js'),
      fileName: 'stories/druxt-debug.stories.js',
    })
    stories.push(resolve(self.options.buildDir, './stories/druxt-debug.stories.js'))
  })
}

export { DruxtNuxtModule }

/**
 * Module options object.
 *
 * @typedef {object} ModuleOptions
 * @property {string} baseUrl - The Base URL of the Drupal JSON:API backend.
 * @property {string} [endpoint=/jsonapi] - The JSON:API endpoint path.
 * @property {object} [proxy] - Proxy settings object.
 * @property {boolean} [proxy.api] - Proxy the JSON:API.
 * @property {(boolean|string)} [proxy.files] - Proxy Drupal's site files directory. Provide String to specify multi-site path.
 *
 * @example @lang js
 * export default {
 *   modules: ['druxt'],
 *   druxt: {
 *     baseUrl: 'https://demo-api.druxtjs.org',
 *     endpoint: '/jsonapi',
 *     proxy: {
 *       api: true,
 *       files: 'default'
 *     }
 *   }
 * }
 */
