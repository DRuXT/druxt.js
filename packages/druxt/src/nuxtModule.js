import { resolve } from 'path'

/**
 * Nuxt module function to install Druxt.
 *
 * @param {ModuleOptions} moduleOptions - DruxtJS module options.
 *
 * @example <caption>Nuxt configuration with module options</caption> @lang js
 * module.exports = {
 *   modules: [
 *     ['druxt', { baseUrl: 'https://demo-api.druxtjs.org' }]
 *   ]
 * }
 *
 * @example <caption>Nuxt configuration with root level options</caption> @lang js
 * module.exports = {
 *   modules: [
 *     'druxt'
 *   ],
 *   druxt: {
 *     baseUrl: 'https://demo-api.druxtjs.org'
 *   }
 * }
 */
const DruxtNuxtModule = function (moduleOptions = {}) {
  const options = {
    ...moduleOptions,
    ...(this.options || {}).druxt
  }

  // Add plugin.
  this.addPlugin({
    src: resolve(__dirname, '../nuxt/plugin.js'),
    fileName: 'druxt.js',
    options
  })

  // Add Vuex plugin.
  this.addPlugin({
    src: resolve(__dirname, '../nuxt/store.js'),
    fileName: 'store/druxt.js',
    options
  })

  // Enable Vuex Store.
  this.options.store = true
}

DruxtNuxtModule.meta = require('../package.json')

export { DruxtNuxtModule }

/**
 * Module options object.
 *
 * @typedef {object} ModuleOptions
 * @property {string} baseUrl - The Base URL of the Drupal JSON:API backend.
 */
