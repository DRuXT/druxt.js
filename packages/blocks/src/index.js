import { DruxtBlocksNuxtModule } from './nuxtModule'

/**
 * The Nuxt.js module function.
 *
 * Installs the module functionality in a Nuxt application.
 *
 * @type {Function}
 * @exports default
 * @name DruxtBlocksModule
 * @see {@link ./nuxtModule|DruxtBlocksModule}
 *
 * @example <caption>nuxt.config.js</caption> @lang js
 * module.exports = {
 *   modules: [
 *     'druxt-blocks'
 *   ],
 *   druxt: {
 *     baseUrl: 'https://demo-api.druxtjs.org'
 *   }
 * }
 */
export default DruxtBlocksNuxtModule

/**
 * Vue.js Mixin.
 *
 * @type {object}
 * @exports DruxtBlocksBlockMixin
 * @see {@link ./mixins/block|DruxtBlocksBlockMixin}
 */
 export { DruxtBlocksBlockMixin } from './mixins/block'
