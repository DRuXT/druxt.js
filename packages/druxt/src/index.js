import { DruxtNuxtModule } from './nuxtModule'

/**
 * DruxtJS JSON:API client.
 *
 * @type {class}
 * @exports DruxtClient
 * @name DruxtClient
 * @see {@link ./class|DruxtClient}
 *
 * @example @lang js
 * import { DruxtClient } from 'druxt'
 * const druxt = new DruxtClient('https://demo-api.druxtjs.org')
 */
export { DruxtClient } from './client'

/**
 * Nuxt module function to install Druxt.
 *
 * @type {Function}
 * @exports default
 * @name DruxtNuxtModule
 * @see {@link ./nuxtModule|DruxtNuxtModule}
 *
 * @example <caption>nuxt.config.js</caption> @lang js
 * module.exports = {
 *   modules: [
 *     'druxt'
 *   ],
 *   druxt: {
 *     baseUrl: 'https://demo-api.druxtjs.org'
 *   }
 * }
 */
export default DruxtNuxtModule

/**
 * Vuex store module.
 *
 * @type {object}
 * @exports DruxtStore
 * @name DruxtStore
 * @see {@link ./stores/druxt|DruxtStore}
 */
export { DruxtStore } from './stores/druxt'

/**
 * DruxtJS utility class.
 *
 * @type {class}
 * @exports DruxtClass
 * @name DruxtClass
 * @see {@link ./class|DruxtClass}
 *
 * @deprecated
 * @private
 */
export { DruxtClass } from './class'
