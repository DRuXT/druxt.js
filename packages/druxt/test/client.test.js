import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { getMockResource } from 'druxt-test-utils'
import mockAxios from 'jest-mock-axios'

import { DruxtClient } from '../src/client'

const baseUrl = 'https://demo-api.druxtjs.org'

jest.mock('axios')

let druxt

describe('DruxtClient', () => {
  beforeEach(() => {
    mockAxios.reset()
    druxt = new DruxtClient(baseUrl)
  })

  test('constructor', () => {
    // Instantiate client without baseUrl.
    expect(() => { return new DruxtClient() }).toThrow('The \'baseUrl\' parameter is required.')

    // Instantiate client with required parameters.
    expect(new DruxtClient(baseUrl)).toBeInstanceOf(DruxtClient)
    expect(mockAxios.create).toHaveBeenCalledWith({ baseURL: baseUrl })

    // Instantiate client with axios headers.
    const headers = { 'X-Druxt': true }
    expect(new DruxtClient(baseUrl, { axios: { headers } })).toBeInstanceOf(DruxtClient)
    expect(mockAxios.create).toHaveBeenCalledWith({ baseURL: baseUrl, headers })
  })

  test('addHeaders', () => {
    expect(druxt.addHeaders()).toBe(false)

    druxt.addHeaders({ 'X-Druxt': true })
    expect(druxt.axios.defaults.headers.common['X-Druxt']).toBe(true)
  })

  test('buildQueryUrl', () => {
    expect(druxt.buildQueryUrl('url', 'query')).toBe('url?query')
    expect(druxt.buildQueryUrl('url', '?query')).toBe('url?query')
    expect(druxt.buildQueryUrl('url', { getQueryString: () => 'query' })).toBe('url?query')
    expect(druxt.buildQueryUrl('url', { query: 'string' })).toBe('url?query=string')

    expect(druxt.buildQueryUrl('url', {})).toBe('url')
    expect(druxt.buildQueryUrl('url', [])).toBe('url')
    expect(druxt.buildQueryUrl('url', Boolean)).toBe('url')
    expect(druxt.buildQueryUrl('url')).toBe('url')
  })

  test('checkPermissions', () => {
    const res = {
      data: {
        meta: {
          omitted: {
            detail: 'Some resources have been omitted because of insufficient authorization.',
            links: {
              help: null,
              item: {
                meta: {
                  detail: 'The current user is not allowed to GET the selected resource. The \'administer node fields\' permission is required.'
                }
              }
            }
          }
        }
      }
    }
    expect(() => druxt.checkPermissions(res)).toThrow('Some resources have been omitted because of insufficient authorization.\n\n Required permissions: administer node fields.')
  })

  test('getCollection', async () => {
    // Get a collection of 'node--page' resources.
    const collection = await druxt.getCollection('node--page')
    expect(mockAxios.get).toHaveBeenLastCalledWith(`${baseUrl}/en/jsonapi/node/page`)
    expect(collection.data.length).toBe(1)

    // Get a filtered collection of 'node--page' resources.
    await druxt.getCollection('node--page', { 'filter[status]': 1 })
    expect(mockAxios.get).toHaveBeenLastCalledWith(`${baseUrl}/en/jsonapi/node/page?filter%5Bstatus%5D=1`)

    // Get a collection with headers set.
    await druxt.getCollection('node--page', {}, { headers: { 'X-Druxt': true }})
    expect(druxt.axios.defaults.headers.common['X-Druxt']).toBe(true)

    // Get collection of nothing.
    const noResource = await druxt.getCollection()
    expect(noResource).toBe(false)
  })

  test('getCollectionAll', async () => {
    // Get all of the 'test--all' resources.
    const query = new DrupalJsonApiParams().addFields('node--recipe', []).addPageLimit(5)
    await druxt.getCollectionAll('node--recipe', query)
    expect(mockAxios.get).toHaveBeenLastCalledWith(`${baseUrl}/en/jsonapi/node/recipe?page%5Boffset%5D=5&page%5Blimit%5D=5&fields%5Bnode--recipe%5D=`)
  })

  test('getIndex', async () => {
    const index = await druxt.getIndex()
    expect(mockAxios.get).toHaveBeenCalledTimes(1)
    expect(mockAxios.get).toHaveBeenCalledWith('/jsonapi')

    expect(Object.keys(index).length).toBe(64)
    expect(index[Object.keys(index)[0]]).toHaveProperty('href')

    const cachedIndex = await druxt.getIndex()
    expect(mockAxios.get).toHaveBeenCalledTimes(1)

    expect(Object.keys(cachedIndex).length).toBe(64)
  })

  test('getIndex - resource', async () => {
    const resourceIndex = await druxt.getIndex('node--page')
    expect(mockAxios.get).toHaveBeenCalledTimes(1)
    expect(mockAxios.get).toHaveBeenCalledWith('/jsonapi')

    expect(resourceIndex).toHaveProperty('href')

    const cachedResourceIndex = await druxt.getIndex('node--page')
    expect(mockAxios.get).toHaveBeenCalledTimes(1)

    expect(cachedResourceIndex).toHaveProperty('href')
  })

  test('getResource', async () => {
    const mockArticle = await getMockResource('node--article')
    const entity = await druxt.getResource(mockArticle.data.type, mockArticle.data.id)
    expect(entity.data).toHaveProperty('type', mockArticle.data.type)

    const error = await druxt.getResource('missing', 'test')
    expect(mockAxios.get).toHaveBeenCalledWith('/jsonapi/missing/test')
    expect(error).toBe(false)

    const empty = await druxt.getResource()
    expect(empty).toBe(false)
  })
})
