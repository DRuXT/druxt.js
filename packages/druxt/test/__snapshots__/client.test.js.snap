// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DruxtClient checkPermissions 1`] = `
Object {
  "detail": "Required permissions:
 - administer node fields",
}
`;

exports[`DruxtClient createResource 1`] = `
Object {
  "detail": "subject: This value should not be null.",
  "source": Object {
    "pointer": "/data/attributes/subject",
  },
  "status": "422",
  "title": "Unprocessable Entity",
}
`;

exports[`DruxtClient getIndex 1`] = `
Object {
  "statusText": "Invalid JSON:API endpoint",
}
`;

exports[`DruxtClient getResource 1`] = `
Object {
  "detail": "The \\"entity\\" parameter was not converted for the path \\"/jsonapi/node/article/{entity}\\" (route name: \\"jsonapi.node--article.individual\\")",
  "links": Object {
    "info": Object {
      "href": "http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.5",
    },
    "via": Object {
      "href": "https://demo-api.druxtjs.org/en/jsonapi/node/article/missing",
    },
  },
  "status": "404",
  "title": "Not Found",
}
`;
